### ISSUE TITLE
Doesn't work with doom?

### ISSUE THREAD

**Issue by dppdppd** [alexluigit/dirvish#1]
`(dirvish)` results in `(file-missing "Cannot open load file" "No such file or directory" "posframe")`

**Comment by alexluigit** [comment]
Hi @dppdppd , thanks for reporting.

I just tried it in doom, and yes, there are some compatability issues with doom's default dired module (maybe there are issues with other modules/packages in doom but I don't know),  

But in terms of your problem, it seems that **posframe** has not been installed properly. How did you install dirvish?

Here's what I put in `~/.doom.d/packages.el`

```
(package! dirvish)
```

And for a temporary fix, try comment out `dired` in your `~/.doom.d/init.el`.

```
:emacs
;; dired 
```
(don't forget to run `doom sync`)

Please tell me if that works, thanks.

**Comment by dppdppd** [comment]
Hey Alex,

I installed it the same way.

So disabling dired seems to have worked, at least on windows.
dired-goto-file brings up a prompt but does nothing. I guess it's a red
herring and I might need to fix some bindings or ignore them. Not sure what
the relationship between dired and dirvish is.

Anyways, thanks for the help.

**Comment by alexluigit** [comment]
> Hey Alex, I installed it the same way. So disabling dired seems to have worked, at least on windows. 

I've added a few patches for doom, now it should work out of box (don't have to disable dired module anymore). 

> dired-goto-file brings up a prompt but does nothing. I guess it's a red herring and I might need to fix some bindings or ignore them. 

If by "does nothing" you mean some UI elements (header/preview/etc.) doesn't update accordingly, this should have been fixed in the latest commit. See scrot below.

![Peek 2021-12-17 21-30](https://user-images.githubusercontent.com/16313743/146560557-e12a7271-aa75-4371-8ae9-6d7204f20bbd.gif)

> Not sure what the relationship between dired and dirvish is. 

`Dired` is a built-in library in Emacs which makes a buffer containing a listing of a files/dirs and provides some commands for directory browsing and file manipulation inside this buffer. `Dirvish` is basically a dired extension to 1. give it a modern UI (icons, file preview, ...) and 2. make some dired commands more intuitive (sorting, pasting, etc.).  The `dired module` in doom-emacs provides a set of sensible default config for dired, it also introduces a few third-party packages and related shims to enhance `Dired`. See https://github.com/hlissner/doom-emacs/blob/develop/modules/emacs/dired/config.el

> Anyways, thanks for the help.

No problem, and thanks for your feedback.
